pkg/reconciler/buildrun/resources/failureDetails.go

```diff
@@ -0,0 +1,115 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	resultErrorMessage         = "error-message"
+	resultErrorReason          = "error-reason"
+	prefixedResultErrorReason  = prefixParamsResultsVolumes + "-" + resultErrorReason
+	prefixedResultErrorMessage = prefixParamsResultsVolumes + "-" + resultErrorMessage
+)
```

Comment:
What is your reasoning behind that? It is currently only used in the resources package.

Reply:
+1 to Baran's assessment. The constants are an implementation detail because we use them on the TaskRun internally but in the BuildRun they are copied into "normal" fields in the status. -> Somebody who uses our Go types does not need to know these constants.